projects/packages/forms/src/blocks/contact-form/components/akismet-panel/index.js

```diff
@@ -0,0 +1,162 @@
+
+		return {
+			isLoading: false,
+			pluginStatus: {
+				isInstalled: !! plugin,
+				isActive: plugin?.status === 'active',
+			},
+		};
+	}, [] );
+
+	const { isInstalled = false, isActive = false } = pluginStatus || {};
+
+	const handleButtonClick = useCallback( () => {
+		const func = isInstalled ? activatePlugin : installAndActivatePlugin;
+		const arg = isInstalled ? 'akismet/akismet' : 'akismet';
+		const isActivationCall = func === activatePlugin;
+
+		setIsInstalling( true );
+		setIsActivating( isActivationCall );
+
+		tracks.recordEvent( 'jetpack_forms_upsell_akismet_click' );
+
+		func( arg ).finally( () => {
+			invalidateResolution( 'getPlugins' );
+			setIsInstalling( false );
+			setIsActivating( false );
+		} );
+	}, [ isInstalled, setIsInstalling, setIsActivating, invalidateResolution, tracks ] );
+
+	return (
+		<PanelBody title={ __( 'Spam protection', 'jetpack-forms' ) } initialOpen={ false }>
+			<div className="akismet-panel__content" aria-live="polite">
+				{ isLoading && (
+					<div className="akismet-panel__status">
+						<div>
+							<Spinner />
+							<span>{ __( 'Checking plugin status…', 'jetpack-forms' ) }</span>
+						</div>
+					</div>
+				) }
+
+				{ ! isLoading && ! pluginStatus && (
+					<div className="akismet-panel__status">
+						<Icon icon="warning" />
+						<span>
+							{ __(
+								'Unable to determine plugin status. Please refresh the page.',
+								'jetpack-forms'
+							) }
+						</span>
+					</div>
+				) }
+
+				{ ! isLoading && pluginStatus && ! isInstalled && (
+					<div className="akismet-panel__panel">
+						<div className="akismet-panel__panel-content">
+							<div>
+								{ createInterpolateElement(
+									__(
+										"Add one-click spam protection for your forms with <a>Akismet</a>. Simply install the plugin and you're set.",
+										'jetpack-forms'
+									),
+									{
+										a: (
+											<a
+												href="https://wordpress.org/plugins/akismet/"
+												target="_blank"
+												rel="noopener noreferrer"
+											/>
+										),
+									}
+								) }
+							</div>
+							<PluginActionButton
+								isInstalling={ isInstalling }
+								isActivating={ isActivating }
+								isInstalled={ isInstalled }
+								onClick={ handleButtonClick }
+							/>
+						</div>
+					</div>
+				) }
+
+				{ ! isLoading && isInstalled && ! isActive && (
+					<div className="akismet-panel__panel">
+						<div className="akismet-panel__panel-content">
+							<div>
+								{ __(
+									"You already have the plugin installed, but it's not activated.",
+									'jetpack-forms'
+								) }
+							</div>
+							<PluginActionButton
+								isInstalling={ isInstalling }
+								isActivating={ isActivating }
+								isInstalled={ isInstalled }
+								onClick={ handleButtonClick }
+							/>
+						</div>
+					</div>
+				) }
+
+				{ ! isLoading && isInstalled && isActive && (
```

Comment:
I'm not entirely sure but we might need to also add "Is configured" check with API keys; I saw some code around which help check if Akismet is configured but I don't remember if the API key is required for Akismet to work.